lib/train/transports/cisco_ios.rb

```diff
@@ -0,0 +1,141 @@
+# encoding: utf-8
+
+require 'train/plugins'
+require 'train/transports/ssh'
+
+module Train::Transports
+  class BadEnablePassword < Train::TransportError; end
+
+  class CiscoIOS < SSH
+    name 'cisco_ios'
+
+    option :host, required: true
+    option :user, required: true
+    option :port, default: 22, required: true
+
+    option :password, required: true
+
+    # Used to elevate to enable mode (similar to `sudo su` in Linux)
+    option :enable_password
+
+    def connection
+      @connection ||= Connection.new(validate_options(@options).options)
+    end
+
+    class Connection < BaseConnection
+      def initialize(options)
+        super(options)
+
+        @session = nil
+        @buf = nil
```

Comment:
[nit] Is there a reason for these? In Ruby, instance variables are `nil` by default, and attempting to access the value of one without having declared it already just returns `nil`. `@session` should only ever be assigned or accessed via `session()`, even. `@buf` may be shared by those two methods, but `run_command_via_connection()` clears it on each invocation anyway.

Reply:
Nope, you make a solid argument. Killing those with fire.